### PR TITLE
Feature IN-47 Seo Support

### PR DESCRIPTION
**Pull request by fork-boy**
## Description
Added Seo layout component to improve SEO support. In this component user enter the page title and content of meta description which is used to improve the SEO of the page. Soon going to add SSR to make SEO more improvised. 

## Screenshot
page_source

## Tests
Done manual test cases, automated tests still need to be done 

### Automated test cases added
  NA

### Manual test cases run
- Entered title and content as props to SEO layout component.

### Browsers tested
- [ ] Chrome
- [ ] Firefox

## Platform dependencies
 NA

### CODE REVIEW

File: components/seo-layout/seo-layout.component.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import Head from 'next/head';
+
+function SeoLayout(props) {
```
Comment (jjalan) [inline components/seo-layout/seo-layout.component.tsx]: Consider using ES6 syntax.

type MetaProps {
  description: string
}

type PageLayoutWithSEOProps {
 title: string
 meta: MetaProps
}

export const PageLayoutWithSEO: React.FC< PageLayoutWithSEOProps > = ({}) => {}

File: pages/index.tsx [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
 
 export default function Home() {
   return (
     <div className={styles.container}>
-      <Head>
-        <title>Create Next App</title>
-        <link rel="icon" href="/favicon.ico" />
-      </Head>
+      <SeoLayout
+        title={'Create Next App'}
```
Comment (jjalan) [inline pages/index.tsx]: Keep all use facing strings in localised file.

Comment (fork-boy) [reply pages/index.tsx]: Rajat will add the localise file using in18

File: components/seo-layout/seo-layout.component.tsx (hunk 1 above)
Comment (fork-boy) [reply components/seo-layout/seo-layout.component.tsx]: done

File: components/seo-layout/page-layout-with-seo.component.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import Head from 'next/head';
+
+type PageLayoutWithSEOProps = {
+  title: string;
+  description: string;
```
Comment (jjalan) [inline components/seo-layout/page-layout-with-seo.component.tsx]:
```
type MetaProps = {
  description: string;
}

type PageLayoutWithSEOProps = {
  title: string;
  meta: MetaProps
}
```

Please scope description to meta tag. I believe I gave the same feedback earlier as well.

File: pages/index.tsx (hunk 2 above)
Comment (jjalan) [reply pages/index.tsx]: Please add a todo in the code.

File: pages/index.tsx [hunk 4]
```diff
@@ -1,17 +1,19 @@
...
   return (
     <div className={styles.container}>
-      <Head>
-        <title>Create Next App</title>
-        <link rel="icon" href="/favicon.ico" />
-      </Head>
+      <PageLayoutWithSEO
+        title={'Create Next App'}
+        description={
+          'This is a page for creating next app for next js introduction'
```
Comment (jjalan) [inline pages/index.tsx]: Please add a TODO in the code that this needs to be localised.

File: pages/index.tsx [hunk 5]
```diff
@@ -1,17 +1,19 @@
...
+        }
+      />
 
       <main className={styles.main}>
         <h1 className={styles.title}>
-          Welcome to <a href="https://nextjs.org">Next.js!</a>
+          Welcome to <a href='https://nextjs.org'>Next.js!</a>
```
Comment (jjalan) [inline pages/index.tsx]: Please add TODO in the code that needs to be localised.

Ex:

```
TODO agentraghav: Localize
```

File: pages/index.tsx (hunk 4 above)
Comment (jjalan) [reply pages/index.tsx]: Please add TODO in the code that needs to be localised.

Ex:

```
TODO agentraghav: Localize
```

File: pages/index.tsx (hunk 2 above)
Comment (jjalan) [reply pages/index.tsx]: Please add TODO in the code that needs to be localised.

Ex:

```
TODO agentraghav: Localize
```

**Review by jjalan — CHANGES_REQUESTED** [review]

**Review by vindru — APPROVED** [review]

**Pull request merged**